### DISCUSSION TITLE
How do I debug the language server

### DISCUSSION THREAD

**Discussion by paulfrench** [eclipse-langium/langium#376] [Q&A] [answered]
After creating a langium project I can debug the extension itself but I cannot workout how to debug the language server? I can see this in the code...

```
  // The debug options for the server
  // --inspect=6009: runs the server in Node's Inspector mode so VS Code can attach to the server for debugging.
  // By setting `process.env.DEBUG_BREAK` to a truthy value, the language server will wait until a debugger is attached.
  const debugOptions = {
    execArgv: [
      "--nolazy",
      `--inspect${process.env.DEBUG_BREAK ? "-brk" : ""}=${
        process.env.DEBUG_SOCKET || "6009"
      }`,
    ],
  };
```

Apologies, this is probably more a vscode question but I'm struggling to find an answer. Any pointers would be appreciated! Thanks

**Comment by msujew** [comment, marked as answer]
Hi @paulfrench,

So basically, the language server is a second process, therefore you need a second debug session/config to actually debug it. We use this `attach` debug config for this:

https://github.com/langium/langium/blob/221b8cff0f48480c2d07de74bcdf81d6c237d339/.vscode/launch.json#L26-L41

We should probably think about adding it to the yeoman generator project...

**Reply by spoenemann** [reply]
See https://github.com/langium/langium/pull/380

**Reply by paulfrench** [reply]
Many thanks for adding. You could rename "Attach" to "Language Server", makes it a bit more obvious what you are debugging. Cheers
